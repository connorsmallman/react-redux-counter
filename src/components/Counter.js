// @flow
import React, { type Node } from 'react';

type Props = {
  increment: Function,
  decrement: Function,
  remove: Function,
  id: string,
  count: number
}

export default function Counter(props: Props): Node {
  const { increment, decrement, remove, count } = props;
  return (
    <div>
      <button onClick={() => decrement(props.id, props.count)}>-</button>
      {count}
      <button onClick={() => increment(props.id, props.count)}>+</button>
      <button onClick={() => remove(props.id)}>Remove</button>
    </div>
  );
}
