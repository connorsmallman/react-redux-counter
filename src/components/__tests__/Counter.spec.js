import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../Counter';

it('renders correctly', () => {
  const tree = renderer.create(
    <Counter count={1} />
  );
  expect(tree).toMatchSnapshot();
});
