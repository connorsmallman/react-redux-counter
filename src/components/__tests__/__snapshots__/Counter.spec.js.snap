// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    -
  </button>
  1
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
